flit_core/flit_core/config.py

```diff
@@ -551,6 +592,18 @@ def read_pep621_metadata(proj, path) -> LoadedConfig:
                 "file or text field required in [project.license] table"
             )
 
+    license_files = []
+    if 'license-files' in proj:
+        _check_type(proj, 'license-files', list)
+        globs = proj['license-files']
+        license_files = _license_files_from_globs(path.parent, globs)
```

Comment:
We should prevent mixing the new-style `license-files` definition with the older `license` table.

> If the new license-files key is present, build tools MUST raise an error if the license key is defined and has a value other than a single top-level string.

```suggestion
    if 'license-files' in proj:
        _check_type(proj, 'license-files', list)
        globs = proj['license-files']
        license_files = _license_files_from_globs(path.parent, globs)
        if isinstance(proj.get('license', None), dict):
            raise ConfigError("license-files replaces license table, do not use both")
```

(The spec also says tools SHOULD add deprecation warnings, but I like to enable the new thing before I start prodding people to move off the old, so let's leave that for after doing a release)